ledger_device_sdk/src/io_new.rs

```diff
@@ -0,0 +1,650 @@
+use crate::seph::{self, PacketTypes};
+
+pub use crate::io_legacy::{ApduHeader, Event, Reply, StatusWords};
+
+use crate::io_callbacks::nbgl_register_callbacks;
+
+#[cfg(any(
+    target_os = "nanox",
+    target_os = "stax",
+    target_os = "flex",
+    target_os = "apex_p"
+))]
+use crate::seph::ItcUxEvent;
+
+use ledger_secure_sdk_sys::seph as sys_seph;
+use ledger_secure_sdk_sys::*;
+
+#[cfg(not(any(target_os = "stax", target_os = "flex", target_os = "apex_p")))]
+use crate::buttons::ButtonEvent;
+#[cfg(not(any(target_os = "stax", target_os = "flex", target_os = "apex_p")))]
+use ledger_secure_sdk_sys::buttons::{get_button_event, ButtonsState};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum CommError {
+    Overflow,
+    IoError,
+}
+
+pub const DEFAULT_BUF_SIZE: usize = 273;
+
+pub struct Comm<const N: usize = DEFAULT_BUF_SIZE> {
+    buf: [u8; N],
+    expected_cla: Option<u8>,
+
+    apdu_type: u8,
+    #[cfg(not(any(target_os = "stax", target_os = "flex", target_os = "apex_p")))]
+    buttons: ButtonsState,
+    // Pending APDU state (set by next_event_ahead callback path). When set, the buffer
+    // currently holds an APDU event that must be consumed before any further io_rx call.
+    pending_apdu: bool,
+    pending_header: ApduHeader,
+    pending_offset: usize,
+    pending_length: usize,
+}
+
+impl<const N: usize> Comm<N> {
+    pub fn new() -> Self {
+        let mut comm = Self {
+            buf: [0; N],
+            expected_cla: None,
+            apdu_type: PacketTypes::PacketTypeNone as u8,
+            #[cfg(not(any(target_os = "stax", target_os = "flex", target_os = "apex_p")))]
```

Comment:
Same as above

Reply:
Done in 8d3bec3512e3678a926280f0b289402aae534277.